### PR TITLE
Cart FO tests with puppeteer - Sanity tests campaign

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Adding sanity tests cart FO
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | TEST_PATH="sanity/cartFO" URL_FO=SHOP_URL npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/15421)

### CODE REVIEW

File: tests/puppeteer/campaigns/sanity/cartFO/01_editCheckCart.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  Add the second product to the cart
+  Check the cart
+  Edit the cart and check it
+ */
+global.scenario('Check the Product page', () => {
+  test('should open the shop page', async () => {
+    await homePage.goTo(global.URL_FO);
+    await homePage.checkHomePage();
+  });
+  test('should go to the first product page', () => homePage.goToProductPage('1'));
```
Comment (SimonGrn) [inline tests/puppeteer/campaigns/sanity/cartFO/01_editCheckCart.js]:
```suggestion
  test('should go to the first product page', async () => await homePage.goToProductPage('1'));
```

And same for every non `async-await` method :)

File: tests/puppeteer/pages/commonPage.js [hunk 2]
```diff
@@ -187,4 +187,16 @@ module.exports = class CommonPage {
...
     await expect(found, `${textValue} was not found as option of select`).to.be.true;
   }
+
+  /**
+   * To get a number from text
+   * @param selector
+   * @return integer
+   */
+  async getNumberFromText(selector) {
+    await this.page.waitFor(2000);
```
Comment (SimonGrn) [inline tests/puppeteer/pages/commonPage.js]: Is it really necessary ? I really don't like fixed wait times...

**Review by SimonGrn — COMMENTED** [review]
Thanks @nesrineabdmouleh

File: tests/puppeteer/campaigns/sanity/cartFO/01_editCheckCart.js (hunk 1 above)
Comment (boubkerbribri) [reply tests/puppeteer/campaigns/sanity/cartFO/01_editCheckCart.js]:
```js
  test('should go to the first product page', async () => {
await homePage.goToProductPage('1'));
});
```
For eslint

Comment (nesrineabdmouleh) [reply tests/puppeteer/campaigns/sanity/cartFO/01_editCheckCart.js]: It causes problems in the esLint :/

File: tests/puppeteer/pages/FO/home.js [hunk 3]
```diff
@@ -21,6 +22,13 @@ module.exports = class Home extends CommonPage {
     await this.page.waitForSelector(this.logoHomePage, {visible: true});
   }
 
+  /**
+   * go to the home page
+   */
+  async goToHomePage() {
+    await this.waitForSelectorAndClick(this.logoHomePage);
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/FO/home.js:29]: Maybe you should wait for navigation here after the click action

File: tests/puppeteer/campaigns/sanity/catalogFO/01_filterProducts.js [hunk 4]
```diff
@@ -28,19 +28,19 @@ global.scenario('Filter Products by categories in Home page', () => {
 
   test('should check and get the products number', async () => {
     await homePage.waitForSelectorAndClick(homePage.allProductLink);
-    allProductsNumber = await homePage.getNumberOfProducts();
+    allProductsNumber = await homePage.getNumberFromText(homePage.totalProducts);
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/sanity/catalogFO/01_filterProducts.js:31]: Maybe add comment here to say that `getNumberFromText` is used to get Number Of Products

File: tests/puppeteer/campaigns/sanity/catalogFO/01_filterProducts.js [hunk 5]
```diff
@@ -28,19 +28,19 @@ global.scenario('Filter Products by categories in Home page', () => {
...
     await expect(allProductsNumber).to.be.above(0);
   });
 
   test('should filter products by the category "Accessories" and check result', async () => {
     await homePage.filterByCategory('6');
-    const numberOfProducts = await homePage.getNumberOfProducts();
+    const numberOfProducts = await homePage.getNumberFromText(homePage.totalProducts);
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/sanity/catalogFO/01_filterProducts.js:37]: same here

File: tests/puppeteer/campaigns/sanity/catalogFO/01_filterProducts.js [hunk 6]
```diff
@@ -28,19 +28,19 @@ global.scenario('Filter Products by categories in Home page', () => {
...
     await expect(numberOfProducts).to.be.below(allProductsNumber);
   });
 
   test('should filter products by the subcategory "Stationery" and check result', async () => {
     await homePage.filterSubCategory('6', '7');
-    const numberOfProducts = await homePage.getNumberOfProducts();
+    const numberOfProducts = await homePage.getNumberFromText(homePage.totalProducts);
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/sanity/catalogFO/01_filterProducts.js:43]: same here

File: tests/puppeteer/campaigns/sanity/cartFO/01_editCheckCart.js [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+    await homePage.goTo(global.URL_FO);
+    await homePage.checkHomePage();
+  });
+  test('should go to the first product page', async () => {
+    await homePage.goToProductPage('1');
+  });
+  test('should add product to the cart', async () => {
+    await productPage.addProductToTheCart();
+  });
+  test('should check that there is one notification', async () => {
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/sanity/cartFO/01_editCheckCart.js]:
```suggestion
  test('should check that number of products in Cart was updated in header', async () => {
```

File: tests/puppeteer/campaigns/sanity/cartFO/01_editCheckCart.js [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+  test('should go to the home page', async () => {
+    await homePage.goToHomePage();
+  });
+  test('should go to the second product page', async () => {
+    await homePage.goToProductPage('2');
+  });
+  test('should add product to the cart', async () => {
+    await productPage.addProductToTheCart();
+  });
+  test('should check that there are two notifications', async () => {
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/sanity/cartFO/01_editCheckCart.js]:
```suggestion
  test('should check that number of products in Cart was updated in header', async () => {
```

**Review by boubkerbribri — COMMENTED** [review]
Thanks @nesrineabdmouleh

File: tests/puppeteer/campaigns/data/FO/cart.js [hunk 9]
```diff
@@ -1,8 +1,15 @@
 module.exports = {
 
   customCartData: {
-    name: 'T-shirt imprimé colibri',
-    quantity: '1',
-    price: '22,94 €',
+    firstProduct: {
+      name: 'T-shirt imprimé colibri',
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/data/FO/cart.js]: You should test in english, so Product name should be in english
```suggestion
      name: 'Hummingbird printed t-shirt',
```

File: tests/puppeteer/campaigns/data/FO/cart.js [hunk 10]
```diff
@@ -1,8 +1,15 @@
...
+      quantity: '1',
+      price: '22,94 €',
+    },
+    secondProduct: {
+      name: 'Pull imprimé colibri',
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/data/FO/cart.js]: Same here
```suggestion
      name: 'Hummingbird printed sweater',
```

File: tests/puppeteer/campaigns/sanity/cartFO/01_editCheckCart.js [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+let page;
+let homePage;
+let productPage;
+let cartPage;
+let totalTTC = 0;
+let itemsNumber = 0;
+
+// creating pages objects in a function
+const init = async () => {
+  page = await global.browser.newPage();
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/sanity/cartFO/01_editCheckCart.js:19]: To test in english
```suggestion
  page = await global.browser.newPage();
  await page.setExtraHTTPHeaders({
    'Accept-Language': 'en-GB'
  });
```

File: tests/puppeteer/campaigns/data/FO/cart.js [hunk 12]
```diff
@@ -1,8 +1,15 @@
 module.exports = {
 
   customCartData: {
-    name: 'T-shirt imprimé colibri',
-    quantity: '1',
-    price: '22,94 €',
+    firstProduct: {
+      name: 'T-shirt imprimé colibri',
+      quantity: '1',
+      price: '22,94 €',
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/data/FO/cart.js]:
```suggestion
      price: '€22.94',
```
Same here

File: tests/puppeteer/campaigns/data/FO/cart.js [hunk 13]
```diff
@@ -1,8 +1,15 @@
...
+    },
+    secondProduct: {
+      name: 'Pull imprimé colibri',
+      quantity: '1',
+      price: '34,46 €',
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/data/FO/cart.js]:
```suggestion
      price: '€34,46',
```
Same here

**Review by matthieu-rolland — APPROVED** [review]
lgtm

**Pull request merged**